src/Shared/TryParseMethodCache.cs

```diff
@@ -31,7 +34,18 @@ public static bool HasTryParseMethod(ParameterInfo parameter)
                     return EnumTryParseMethod.MakeGenericMethod(type);
                 }
 
-                var staticMethods = type.GetMethods(BindingFlags.Public | BindingFlags.Static);
+                if (TryGetDateTimeTryPareMethod(type, out var methodDateInfo))
+                {
+                    return methodDateInfo;
+                }
+
+                if (TryGetNumberStylesTryGetMethod(type, out var methodInfo, out var _))
+                {
+                    return methodInfo;
+                }
+
+                var staticMethods = type.GetMethods(BindingFlags.Public | BindingFlags.Static)
+                    .OrderByDescending(m => m.GetParameters().Length); ;
```

Comment:
I don't think we need any OrderBy logic. I think we can just check for the 3-argument `IFormatProvider` overload of TryParse using `GetMethod` first, then check for the 2-argument overload second.